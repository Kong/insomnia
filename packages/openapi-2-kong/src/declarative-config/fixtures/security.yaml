openapi: 3.0.0

info:
  title: Security Example
  version: 1.0.0

servers:
  - url: http://backend.com/path

paths:
  /only/basic:
    get:
      security:
        - really_basic: []
      responses:
        '200':
          description: Saying hello

  /only/key:
    get:
      security:
        - my_api_key: []
      responses:
        '200':
          description: Saying hello

  /only/oath2:
    get:
      security:
        - petstore_oauth2:
            - read:pets
      responses:
        '200':
          description: Saying hello

  /only/oidc:
    get:
      security:
        - petstore_openid:
            - write:pets
      responses:
        '200':
          description: Saying hello

  /and/based/auth:
    get:
      security:
        - petstore_oauth2:
            - read:pets
          #my_api_key: []
      responses:
        '200':
          description: Saying hello

  /or/based/auth:
    get:
      security:
        - my_api_key: []
      #- really_basic: []
      responses:
        '200':
          description: Saying hello

components:
  securitySchemes:
    really_basic:
      type: http
      scheme: basic
      x-kong-security-basic-auth:
        tags:
          - should match
        config:
          hide_credentials: true
    my_api_key:
      type: apiKey
      name: api_key_by_me
      in: header
      x-kong-security-basic-auth:
        tags:
          - should not match
    your_api_key:
      type: apiKey
      name: api_key_by_you
      in: header
    petstore_oauth2:
      type: oauth2
      flows:
        clientCredentials:
          tokenUrl: http://example.org/api/oauth/dialog
          scopes:
            write:pets: modify pets in your account
            read:pets: read your pets
    petstore_openid:
      type: openIdConnect
      openIdConnectUrl: http://example.org/oid-discovery
