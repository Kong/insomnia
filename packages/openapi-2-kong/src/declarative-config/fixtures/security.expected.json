{
  "services": [
    {
      "protocol": "http",
      "host": "Security_Example",
      "port": 80,
      "path": "/path",
      "name": "Security_Example",
      "plugins": [],
      "routes": [
        {
          "methods": ["GET"],
          "strip_path": false,
          "tags": ["OAS3_import", "OAS3file_security.yaml"],
          "name": "Security_Example-only_oath2-get",
          "plugins": [
            {
              "config": { "auth_methods": ["client_credentials"] },
              "name": "openid-connect",
              "tags": ["OAS3_import", "OAS3file_security.yaml"]
            }
          ],
          "paths": ["/only/oath2$"]
        },
        {
          "methods": ["GET"],
          "strip_path": false,
          "tags": ["OAS3_import", "OAS3file_security.yaml"],
          "name": "Security_Example-only_key-get",
          "plugins": [
            {
              "config": { "key_names": ["api_key_by_me"] },
              "name": "key-auth",
              "tags": ["OAS3_import", "OAS3file_security.yaml"]
            }
          ],
          "paths": ["/only/key$"]
        },
        {
          "methods": ["GET"],
          "strip_path": false,
          "tags": ["OAS3_import", "OAS3file_security.yaml"],
          "name": "Security_Example-and_based_auth-get",
          "plugins": [
            {
              "config": { "auth_methods": ["client_credentials"] },
              "name": "openid-connect",
              "tags": ["OAS3_import", "OAS3file_security.yaml"]
            }
          ],
          "paths": ["/and/based/auth$"]
        },
        {
          "methods": ["GET"],
          "strip_path": false,
          "tags": ["OAS3_import", "OAS3file_security.yaml"],
          "name": "Security_Example-or_based_auth-get",
          "plugins": [
            {
              "config": { "key_names": ["api_key_by_me"] },
              "name": "key-auth",
              "tags": ["OAS3_import", "OAS3file_security.yaml"]
            }
          ],
          "paths": ["/or/based/auth$"]
        },
        {
          "methods": ["GET"],
          "strip_path": false,
          "tags": ["OAS3_import", "OAS3file_security.yaml"],
          "name": "Security_Example-only_oidc-get",
          "plugins": [
            {
              "config": {
                "scopes_required": ["write:pets"],
                "issuer": "http://example.org/oid-discovery"
              },
              "name": "openid-connect",
              "tags": ["OAS3_import", "OAS3file_security.yaml"]
            }
          ],
          "paths": ["/only/oidc$"]
        },
        {
          "methods": ["GET"],
          "strip_path": false,
          "tags": ["OAS3_import", "OAS3file_security.yaml"],
          "name": "Security_Example-only_basic-get",
          "plugins": [
            {
              "name": "basic-auth",
              "tags": ["OAS3_import", "OAS3file_security.yaml", "should match"],
              "config": { "hide_credentials": true }
            }
          ],
          "paths": ["/only/basic$"]
        }
      ],
      "tags": ["OAS3_import", "OAS3file_security.yaml"]
    }
  ],
  "upstreams": [
    {
      "tags": ["OAS3_import", "OAS3file_security.yaml"],
      "name": "Security_Example",
      "targets": [
        {
          "target": "backend.com:80",
          "tags": ["OAS3_import", "OAS3file_security.yaml"]
        }
      ]
    }
  ],
  "_format_version": "1.1"
}
